neo/Network/P2P/Connection.cs

```diff
@@ -66,16 +68,24 @@ private void WsReceive()
                 failure: ex => new Tcp.ErrorClosed(ex.Message));
         }
 
-        public void Disconnect(bool abort = false)
+        protected void Disconnect(DisconnectReason reason, string message = "", byte[] data = null)
         {
             disconnected = true;
+
+            var payload = DisconnectPayload.Create(reason, message, data);
+            var disconnectMessage = Message.Create(MessageCommand.Disconnect, payload);
+
             if (tcp != null)
             {
-                tcp.Tell(abort ? (Tcp.CloseCommand)Tcp.Abort.Instance : Tcp.Close.Instance);
+                tcp.Tell(Tcp.Write.Create((ByteString.FromBytes(disconnectMessage.ToArray()))));
+                tcp.Tell(Tcp.Close.Instance);
```

Comment:
What happened to `Abort`?

Reply:
Replaced by Tcp.Close, as Tcp.Close can flush pending writes. 😂

Reply:
In fact, previously, I think that one of them was almost never being used. I checked this sometime ago, I do not remember if it was `Abort` or `Close`.

Reply:
I think that `Close` was never being used. It was all `Abort`, right?

![image](https://user-images.githubusercontent.com/17395762/66693217-0a563600-ec7d-11e9-9775-cec205230ca1.png)

Reply:
And we have never used `Tcp.Abort` actually.

Reply:
For me it looks good, @Tommo-L, I agree. `Abort` could be used when force exiting, something like that.

Reply:
> I think that `Close` was never being used. It was all `Abort`, right?
> 
> ![image](https://user-images.githubusercontent.com/17395762/66693217-0a563600-ec7d-11e9-9775-cec205230ca1.png)

Yes, sorry, it's all `Abort`. Maybe we should add it back.

Reply:
I want to change here. After received the ack of `disconnect` message back, then it can abort the connection.

Reply:
done